tidb-functions: update tidb-functions.md doc (#8777)
pingcap/docs: functions-and-operators/tidb-functions.md

| [`TIDB_VERSION()`](#tidb_version) | The `TIDB_VERSION` function returns the TiDB version with additional build information. |
| [`TIDB_DECODE_SQL_DIGESTS(digests, stmtTruncateLength)`](#tidb_decode_sql_digests) | The `TIDB_DECODE_SQL_DIGESTS()` function is used to query the normalized SQL statements (a form without formats and arguments) corresponding to the set of SQL digests in the cluster. |
| `VITESS_HASH(str)` | The `VITESS_HASH` function returns the hash of a string that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |

> | `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a SHARD INDEX to scatter the index hotspot. A SHARD INDEX is an expression index with a `TIDB_SHARD`function as the prefix.|

Reviewer: If SHARD INDEX is natural language, shall we use the lower case "shard index"? Or wrap it with backticks.
```suggestion
| `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a SHARD INDEX to scatter the index hotspot. A SHARD INDEX is an expression index with a `TIDB_SHARD` function as the prefix.|
```

> | `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index with a `TIDB_SHARD`function as the prefix.|

Reviewer: ```suggestion
| `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index with a `TIDB_SHARD` function as the prefix.|
```

- [`Statement Summary Tables`](/statement-summary-tables.md)
- [`INFORMATION_SCHEMA.TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)

### TIDB_SHARD

The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index prefixed with a `TIDB_SHARD` function.

#### Shard index

- Creation:

>     When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.

Author: ```suggestion
    You can use `uk((tidb_shard(a)), a))` to create a shard index for the index `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```

>     When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.

Reviewer: See https://github.com/pingcap/docs-cn/pull/8775/files#r890124259

Author: `uk((tidb_shard(a)), a))` 就是创建的方式，我加了一句 You can use `uk((tidb_shard(a)), a))` to create a shard index for the index `a`.

> - Scenario:

Reviewer: ```suggestion
- Scenarios:
```

>     You can use `uk((tidb_shard(a)), a))` to create a shard index for the index `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.

Author: ```suggestion
    You can use `uk((tidb_shard(a)), a))` to create a shard index for the field `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```

>     You can use `uk((tidb_shard(a)), a))` to create a shard index for the field `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.

Reviewer: ```suggestion
    To create a shard index for the index field `a`, you can use `uk((tidb_shard(a)), a))`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the scalability of the cluster.
```

>     - When there is a write hotspot caused by monotonically increasing or decreasing keys on the unique secondary index, and the index contains integer type fields.

Reviewer: ```suggestion
    - There is a write hotspot caused by monotonically increasing or decreasing keys on the unique secondary index, and the index contains integer type fields.
```

    - The SQL statement executes an equality query based on all fields of the secondary index, either as a separate `SELECT` or as an internal query generated by `UPDATE`, `DELETE`, etc. The equality query includes `a = 1` or `a IN (1, 2, ......)` these two ways.

- Limitations:

>     - SHARD INDEX cannot be used with inequality or `IN` queries.

Author: ```suggestion
    - SHARD INDEX cannot be used with inequality queries.
```

>     - SHARD INDEX cannot be used in inequality queries.

Reviewer: ```suggestion
    - SHARD INDEX cannot be used in inequality or `IN` queries.
```

Author: Remove `IN` query in https://github.com/pingcap/docs-cn/pull/9632

    - SHARD INDEX cannot be used with an outmost `AND` operator mixed with `OR`.

>     - SHARD INDEX cannot be used with `GROUP BY` clause.

Reviewer 2: ```suggestion
    - SHARD INDEX cannot be used in`GROUP BY` clause.
```

    - SHARD INDEX cannot be used with `ORDER BY` clause.

>     - SHARD INDEX cannot be used with the `ON` clause.

Reviewer 2: change 'with' to 'in'

>     - SHARD INDEX cannot be used in `GROUP BY` clause.
>     - SHARD INDEX cannot be used in `ORDER BY` clause.
>     - SHARD INDEX cannot be used in the `ON` clause.

Reviewer: ```suggestion
    - SHARD INDEX cannot be used in the `GROUP BY` clause.
    - SHARD INDEX cannot be used in the `ORDER BY` clause.
    - SHARD INDEX cannot be used in the `ON` clause.
```

    - Cannot be used in the `WHERE` subquery.

>     - Can only be used to scatter unique indexes of integer fields.

Reviewer: ```suggestion
    - Can be used to scatter unique indexes of only the integer fields.
```

>     - SHARD INDEX might be ineffective in composite indexes.

Reviewer: ```suggestion
    - SHARD INDEX might take effect in composite indexes.
```

    - Shard index cannot go through FastPlan process, which affects optimizer performance.

>     - Shard index cannot be used in preparing the execution plan cache.

Reviewer: Because all items start with "Shard index", it is suggested to remove the common subject.

Reviewer: Cannot be used to ...

#### Synopsis

```ebnf+diagram
TIDBShardExpr ::=
    "TIDB_SHARD" "(" expr ")"
```

#### Example

- Use the `TIDB_SHARD` function to calculate the SHARD value.

    The following command shows how to use the `TIDB_SHARD` function to calculate the SHARD value of `12373743746`:

    {{< copyable "sql" >}}

    ```sql
    SELECT TIDB_SHARD(12373743746);
    ```

- The SHARD value is:

    ```sql
    +-------------------------+
    | TIDB_SHARD(12373743746) |
    +-------------------------+
    |                     184 |
    +-------------------------+
    1 row in set (0.00 sec)
    ```

> - Create a SHARD INDEX with the `TIDB_SHARD` function:

Reviewer: ```suggestion
- Create a SHARD INDEX using the `TIDB_SHARD` function:
```
